### ISSUE TITLE
Add support for comments 

### ISSUE THREAD

**Issue by lovasoa** [ms705/nom-sql#30]
It would be nice to be able to parse SQL statements with comments in them.

#### Example

```sql
create--blah blah
table--blah blah
tt--blah blah
(--blah blah
x--blah blah
)--blah blah
;
```

**Comment by ms705** [comment]
Agreed -- and support for `/* ... */`-style comments would also be nice.

The best way to do this is perhaps to pre-process the entire input string with a parse rule (or regex) that removes all comments before running through the normal parser sequence.

I'll try to implement this when I get a chance -- probably not for another two weeks or so, though.

**Comment by lovasoa** [comment]
I'm not sure this can be done with a simple input preprocessing rule. How would you handle cases such as : 

```sql
select '
   -- this is not an SQL comment
';
```

or

```sql
INSERT INTO t VALUES ('/*', '*/');
```

or

```sql
se/**/lect 0;
```
